### ISSUE TITLE
runAsUser, runAsGroup, fsGroup

### ISSUE THREAD

**Issue by cimnine** [vshn/k8up#480]
## Summary

**As** an ops engineer setting up k8up\
**I want** control over the UID and GID with which (w)restic is started\
**So that** I can ensure tight security.

## Context

Currently restic is started as a specified user. This is either 1001 (65532 after #474) or root (maybe not anymore after #474), because these UIDs are burnt into the Docker image in the `Dockerfile`.

This could lead to problems with the mounts. I.e. if a Restore is performed, it will create the files with the permissions of the user it runs. When the container, who's files were restored, is restarted, it will see files with the UID 1001 (or 65532 or 0 respectively), which may prevent it's process to modify the files (edit, delete, maybe even read).

## Out of Scope

* Changes to other features

## Further links

* https://kubernetes.io/blog/2020/12/14/kubernetes-release-1.20-fsgroupchangepolicy-fsgrouppolicy/
* https://kubernetes.io/docs/tasks/configure-pod-container/security-context/

## Acceptance criteria

* It is possible to set the UID/GID with which the restore Pod is run.
* The GID of the mount `/restore` matches the GID with which the restore Pod is run.

## Implementation Ideas

* A global flag (e.g. env var, CLI flag, both) to define the default UID and GID for restore Pods, plus
* a flag in the Restore CRD to define the UID and GID under which the restore Pod is created.

**Comment by ccremer** [comment]
This is kinda a duplicate of #263 as they both aim to solve the same problem.
The current workaround is that there are image variants with root and non-root users.

However, since #497 removes the root user variants, we need an alternative to keep feature parity.
I'd suggest going forward with #263 instead of the idea in this issue for better flexibility (sometimes customizing UID and GID are not enough and I'd rather move this config to the CRD than CLI/ENV flags)

**Comment by cimnine** [comment]
Closed in favor of #263
